rfcs/convergence/authoring-stories.md

```diff
@@ -0,0 +1,290 @@
+
+## Problem statement
+
+For convergence (vNext) we agreed on a collocated stories approach that uses [Component Story Format (CSF)](https://storybook.js.org/docs/react/api/csf) for implementation.
+
+**What are we missing:**
+
+1. what should we use for interactive props playground - controls or knobs?
+2. how to use `argTypes`
+3. should controls work for all stories or only for general/default one
+4. should we render descriptions table in Canvas Control pane
+5. how to author e2e suites for those stories
+6. how to properly annotate stories with TS metadata to get the best DX possible
+7. how structure stories in the storybook nav tree / @miroslavstastny
+8. what to do if story files are getting way too big (dissecting big story files into smaller ones) / @ling1726
+
+> **💡 NOTE:**
+> This RFC will address the first 4 points for now (should be updated with others later).
+
+## Detailed Design or Proposal
+
+### 1. what should we use for interactive props playground - controls or knobs?
+
+**Proposal:**
+
+We should use [controls](https://storybook.js.org/docs/react/essentials/controls).
+
+**Why:**
+
+- while there are some limitations with using controls (they don't mirror the state in URL thus cannot be tested in e2e scenarios), using them is an industry standard in the open source world, also knobs will be deprecated in the next major storybook versions.
+
+### 2. how to use `argTypes`
+
+A pattern that is used in some of the converged stories is to define everything by hand:
+
+```ts
+AccordionExample.argTypes = {
+  inline: {
+    defaultValue: false,
+    control: 'boolean',
+  },
+  navigable: {
+    defaultValue: false,
+    control: 'boolean',
+  },
+  circular: {
+    defaultValue: false,
+    control: 'boolean',
+  },
+  multiple: {
+    defaultValue: false,
+    control: 'boolean',
+  },
+  // ... other definitions of controls
+};
+```
+
+**Proposal:**
+
+- this is not necessary as storybook generates all those controls automatically from TS metadata (Props interface)
+- `argTypes` can be used for use cases when:
+
+  - SB is unable to generated appropriate controls
+  - we wanna override default value based on API
+  - we wanna hide some controls that don't make sense to be handled in particular story
+
+  ```tsx
+  const StoryName = (props: {defaultOpen?:boolean}) => { /* ... */ }
+  // HIDE actionable Control
+  StoryName.argTypes = {
+    defaultOpen: {
+      control: false,
+    },
+  },
+  ```
+
+### 3. should controls work for all stories or only for general/default one
+
+**Proposals (2)**
+
+_1. ✅ provide controls with control pane only for default/playground story_
+
+> This is preferred approach - Based on feedback from @teams-prg/@cxe-prg team
+
+With that approach we would probably need to completely get rid of `controls` addon pane from all stories except `Docs` view and our `Default/Playground` story.
```

Comment:
I'd propose that the default story should _not_ have any controls, and instead have some preset examples showing basic usage of the control. Only the playground story(ies) should have controls -- they're useful if you want to try out more of the functionality of the component, but it's better to have some functional examples to start with on the default story, for someone learning a new control library.

Reply:
I think the Default/Plaground story should be one and the same thing. We need the simplest variation of this component to let the user test as many props as possible.